### PR TITLE
Allow to install PrestaShop without allow_url_fopen enabled

### PR DESCRIPTION
**Pull request by rGaillard**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | When you try to install PrestaShop with allow_url_fopen disabled, it will fail
| Type?         | bug fix
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Please indicate how to best verify that this PR is correct.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9355)

### CODE REVIEW

**Review by eternoendless — DISMISSED** [review]
Ok for this temporary workaround.

Real fix must be done to stream the download instead of loading in memory: http://forge.prestashop.com/browse/BOOM-6033

File: classes/Language.php [hunk 1]
```diff
@@ -1089,9 +1089,10 @@ public static function downloadXLFLanguagePack($locale, &$errors = array(), $typ
         if (!is_writable(dirname($file))) {
             // @todo Throw exception
             $errors[] = Context::getContext()->getTranslator()->trans('Server does not have permissions for writing.', array(), 'Admin.International.Notification').' ('.$file.')';
+        } elseif ($content = Tools::file_get_contents($url)) {
+            file_put_contents($file, $content);
         } else {
-            $fs = new Filesystem();
-            $fs->copy($url, $file, true);
+            $errors[] = Context::getContext()->getTranslator()->trans('Language pack unavailable %s', array($url), 'Admin.International.Notification');
```
Comment (LouiseBonnard) [inline classes/Language.php]: Crowdin still has the string _Language pack unavailable._ under the Admin.International.Notification domains.

**Review by eternoendless — DISMISSED** [review]

File: classes/Language.php (hunk 1 above)
Comment (eternoendless) [reply classes/Language.php]: Thanks for checking @LouiseBonnard ! 👍

Comment (rGaillard) [reply classes/Language.php]: If we remove the url, the error is less verbose :(

**Comment by marionf** [comment]
@rGaillard 

We can't install prestashop with allow_url_fopen disabled

capture d ecran_71

**Comment by rGaillard** [comment]
@LouiseBonnard better like this?

**Comment by LouiseBonnard** [comment]
Yes !

**Comment by rGaillard** [comment]
@marionf ok, i understand. The problem is how we check the configuration variables with ini_get(). Put "php_admin_value allow_url_fopen false" in your apache configuration and you will be able to reproduce.

ini_get('allow_url_fopen') will return "false" so it will equal to true ;)

**Comment by PierreRambaud** [comment]
Agreed with @rGaillard, FYI using
`php_admin_value allow_url_fopen false` return a string `"false"`  when 
`php_admin_flag allow_url_fopen false` return a string `"0"`

**Comment by rGaillard** [comment]
@eternoendless @PierreRambaud Also, i think that we can remove that url_fopen is mandatory (since 1.7.2.2) on the system step. The legacy is using Tools::file_get_contents() which is using curl by default and now Guzzle (which use curl too) is used.

**Comment by PierreRambaud** [comment]
But regarding the problem, the patch should maybe be:
```php
diff --git a/classes/ConfigurationTest.php b/classes/ConfigurationTest.php
index 7879404c55..6130c6fbe5 100644
--- a/classes/ConfigurationTest.php
+++ b/classes/ConfigurationTest.php
@@ -176,7 +176,7 @@ class ConfigurationTestCore
 
     public static function test_fopen()
     {
-        return ini_get('allow_url_fopen');
+        return in_array(ini_get('allow_url_fopen'), array('On', 'on', '1'));
     }
 
     public static function test_system($funcs)
```

Wdyt @rGaillard?

**Comment by PierreRambaud** [comment]
Two solutions, keeping allow_url_fopen but we need to change the test_fopen function:
```php
diff --git a/classes/ConfigurationTest.php b/classes/ConfigurationTest.php
index 7879404c55..6130c6fbe5 100644
--- a/classes/ConfigurationTest.php
+++ b/classes/ConfigurationTest.php
@@ -176,7 +176,7 @@ class ConfigurationTestCore
 
     public static function test_fopen()
     {
-        return ini_get('allow_url_fopen');
+        return in_array(ini_get('allow_url_fopen'), array('On', 'on', '1'));
     }
 
     public static function test_system($funcs)
```
Or as you said, remove this mandatory function :)

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
We need to decide if we remove `allow_url_fopen` or if we improve `test_fopen` :)

**Review by PierreRambaud — APPROVED** [review]
Changes are added on #9361

**Comment by PierreRambaud** [comment]
Thanks @rGaillard  ;)

**Pull request merged**